sns/cm/repair/ut/xform.c

```diff
@@ -438,6 +438,87 @@ static void cp_multi_failures_post(char data, int cnt, int index)
 	m0_semaphore_down(&sem);
 }
 
+static void buf_initialize(struct m0_buf *buf, uint32_t size, uint32_t len)
+{
+	uint32_t j;
+	uint8_t	*arr;
+
+	for (j = 0; j < size; ++j) {
+		M0_ALLOC_ARR(arr, len);
+		M0_UT_ASSERT(arr != NULL);
+		m0_buf_init(&buf[j], arr, len);
+	}
+}
+
+static void buf_free(struct m0_buf *buf, uint32_t count)
+{
+	uint32_t i;
+
+	for (i = 0; i < count; ++i)
+		m0_buf_free(&buf[i]);
+}
+
+static void bufvec2buf(struct m0_buf *buf, struct m0_bufvec *bufvec, uint32_t count)
```

Comment:
I remember these functions are defined somewhere already, aren't they?

Reply:
Yes. Those are static functions in parity_math.c.

Reply:
Duplicated code is hard to maintain.

Please reuse those functions by exporting them from there and using them here.

Reply:
Removed these static functions and used `m0_bufvec_to_data_copy()` and `m0_data_to_bufvec_copy()` functions.